### PR TITLE
fix(webpack): add cacheProvider for Linaria v3

### PR DESCRIPTION
**Pull request by malash**
## Motivation

This PR add the same feature as https://github.com/callstack/linaria/pull/884 for Linaria v3.

## Summary

See https://github.com/callstack/linaria/pull/884

## Test plan

I tested this PR in https://github.com/malash/linaria-issue-881/commit/ae8ecca1d8adb812c39ede9c897d405549fa5ae4 with `cacheProvider: undefined` and `cacheProvider: require.resolve("./linariaFileCache")` and they work as expected.

@Anber

### CODE REVIEW

File: packages/webpack4-loader/src/index.ts [hunk 1]
```diff
@@ -137,16 +141,21 @@ export default function webpack4Loader(
...
-    this.callback(
-      null,
-      `${result.code}\n\nrequire(${stringifiedRequest});`,
-      castSourceMap(result.sourceMap)
-    );
+    getCacheInstance(cacheProvider)
+      .then((cacheInstance) => cacheInstance.set(this.resourcePath, cssText))
+      .then(() => {
+        const request = `${outputFileName}!=!${outputCssLoader}?cacheProvider=${encodeURIComponent(
+          cacheProvider ?? ''
```
Comment (malash) [inline packages/webpack4-loader/src/index.ts:148]: I also fixed this line in https://github.com/callstack/linaria/pull/890

File: packages/webpack5-loader/src/index.ts [hunk 2]
```diff
@@ -130,16 +134,21 @@ export default function webpack5Loader(
...
-    this.callback(
-      null,
-      `${result.code}\n\nrequire(${stringifiedRequest});`,
-      result.sourceMap ?? undefined
-    );
+    getCacheInstance(cacheProvider)
+      .then((cacheInstance) => cacheInstance.set(this.resourcePath, cssText))
+      .then(() => {
+        const request = `${outputFileName}!=!${outputCssLoader}?cacheProvider=${encodeURIComponent(
+          cacheProvider ?? ''
```
Comment (malash) [inline packages/webpack5-loader/src/index.ts:141]: I also fixed this line in https://github.com/callstack/linaria/pull/890

File: packages/webpack5-loader/src/index.ts [hunk 3]
```diff
@@ -46,7 +50,7 @@ export default function webpack5Loader(
   const webpackResolveOptions = this._compilation?.options.resolve;
 
   // Resolved configuration contains empty list of extensions as a default value
-  // https://github.com/callstack/linaria/issues/855
+  // https://github.com/callstack/linaria/issue/Users/malash/Projects/linaria/packages/webpack4-loader/lib/cache.jss/855
```
Comment (Anber) [inline packages/webpack5-loader/src/index.ts:53]: Accidental cmd+c? :)

Comment (malash) [reply packages/webpack5-loader/src/index.ts]: Yes, reverted.

**Review by Anber — APPROVED** [review]

**Pull request merged**